### DISCUSSION TITLE
socket.io-client typescript error

### DISCUSSION THREAD

**Discussion by gwendalF** [socketio/socket.io#4224] [Q&A] [answered]
Hello, I am starting a new project with socketio-client. I use the version 4.4.0 but I have troubles with Typescript. When I try to use `socket.on('event')` I get the following error:
My socket was create by the following:
`export const socket = io('http://localhost:5000');`

Property 'on' does not exist on type 'Socket<DefaultEventsMap, DefaultEventsMap>'.ts(2339)

How can I do to be able to use the on function on my socket? I tried the Typescript definition but it did not change anything.

Thanks

**Comment by darrachequesne** [comment, marked as answer]
Hi! That's weird, are you able to reliably reproduce? Could you please check with the example [there](https://github.com/socketio/socket.io-fiddle/tree/typescript)?

See also: https://socket.io/docs/v4/typescript/

**Comment by gwendalF** [comment]
Hello,
It may have been a cache issue. I cannot reproduce the error and the example working fine
Thanks

**Comment by anthonyalayo** [comment]
I am hitting the same issue.
